File: compiler/src/main/java/jp/takuji31/metro/MetroCompiler.java
package jp.takuji31.metro;

import com.google.auto.common.BasicAnnotationProcessor;
import com.google.auto.service.AutoService;

import java.util.Arrays;

import javax.annotation.processing.Processor;
import javax.annotation.processing.SupportedAnnotationTypes;
import javax.annotation.processing.SupportedSourceVersion;
import javax.lang.model.SourceVersion;

import jp.takuji31.metro.annotations.APIClient;

@AutoService(Processor.class)
@SupportedSourceVersion(SourceVersion.RELEASE_8)
public class MetroCompiler extends BasicAnnotationProcessor {
    @Override
    protected Iterable<? extends ProcessingStep> initSteps() {
        return Arrays.asList(new CreateRequestBuilderStep(processingEnv));
    }
}

File: compiler/src/main/java/jp/takuji31/metro/MetroCompiler.kt
package jp.takuji31.metro

import com.google.auto.common.BasicAnnotationProcessor
import com.google.auto.service.AutoService
import javax.annotation.processing.Processor
import javax.annotation.processing.SupportedSourceVersion
import javax.lang.model.SourceVersion

@AutoService(Processor::class)
@SupportedSourceVersion(SourceVersion.RELEASE_8)
class MetroCompiler : BasicAnnotationProcessor() {
    override fun initSteps(): Iterable<BasicAnnotationProcessor.ProcessingStep> {
        return mutableListOf(CreateRequestBuilderStep(processingEnv))
    }
}
